### PR TITLE
feat: add metrics data points

### PR DESCRIPTION
**Pull request by Repraance**
**What kind of change does this PR introduce?**

**Did you add tests(unit,e2e) for your changes?**

**Does this PR introduce a breaking change?**

**What needs to be documented once your changes are merged?**

### CODE REVIEW

File: packages/platform-web/src/node/features/html-render/lib/getPageMiddleware.ts [hunk 1]
```diff
@@ -10,21 +10,32 @@ import { IHandlePageRequest, RequestContext, ISendHtml } from '../serverHooks';
...
 function createPageHandler(serverPluginContext: IServerPluginContext) {
+  const {
+    traces: { serverRequestTrace }
+  } = serverPluginContext;
   const wrappedSendHtml = async (
     html: string,
     { req, res }: RequestContext
   ) => {
+    const sendHtmlOriginalTrace = serverRequestTrace.traceChild(
+      'SHUVI_SERVER_SEND_HTML_ORIGINAL'
```
Comment (liximomo) [inline packages/platform-web/src/node/features/html-render/lib/getPageMiddleware.ts:21]: We should define all these `SHUVI_SERVER_SEND_HTML_ORIGINAL` as constants in a place, So we can get a clear picture of all traces we have.

**Review by liximomo — CHANGES_REQUESTED** [review]

File: packages/platform-web/src/shuvi-app/react/AppContainer.tsx [hunk 2]
```diff
@@ -22,12 +24,52 @@ function ErrorGuard({ children = null }: React.PropsWithChildren<{}>) {
...
+      navigationTrace.current = clientRenderTrace.traceChild(
+        'SHUVI_NAVIGATION_DONE'
+      );
+      const navigationTriggeredTrace = clientRenderTrace.traceChild(
+        'SHUVI_NAVIGATION_TRIGGERED'
+      );
+      const fromPath = `${from.pathname}${from.search}`;
+      const toPath = `${to.pathname}${to.search}`;
+      const navigationId = uuid();
+      navigationTrace.current.setAttribute('from', fromPath);
```
Comment (liximomo) [inline packages/platform-web/src/shuvi-app/react/AppContainer.tsx:50]: I would like to know which trace has what attributes, maybe we should pre-define all of these in a place.

**Review by liximomo — CHANGES_REQUESTED** [review]

File: packages/platform-web/src/node/features/html-render/lib/getPageMiddleware.ts (hunk 1 above)
Comment (Repraance) [reply packages/platform-web/src/node/features/html-render/lib/getPageMiddleware.ts]: Done.
In fact, it was placed in constants at the first version, and then there were some field adjustments later, and I found that next.js did not store them in constants, so I deleted it 😂

File: packages/platform-web/src/node/features/html-render/lib/getPageMiddleware.ts [hunk 3]
```diff
@@ -69,7 +100,25 @@ export async function getPageMiddleware(
...
+        SHUVI_SERVER_RUN_PAGE_MIDDLEWARE.attrs.error.name,
+        false
+      );
+      runPageMiddlewareTrace.setAttribute(
+        SHUVI_SERVER_RUN_PAGE_MIDDLEWARE.attrs.statusCode.name,
+        res.statusCode
+      );
+      runPageMiddlewareTrace.stop();
     } catch (error) {
+      runPageMiddlewareTrace.setAttribute(
```
Comment (liximomo) [inline packages/platform-web/src/node/features/html-render/lib/getPageMiddleware.ts:113]: It looks like it better to have a `setAttributes` method. e.g.

```
unPageMiddlewareTrace.setAttributes({
   'a': 1
   'b': 2
})
```

**Review by liximomo — CHANGES_REQUESTED** [review]

File: packages/platform-web/src/node/features/html-render/lib/getPageMiddleware.ts (hunk 3 above)
Comment (Repraance) [reply packages/platform-web/src/node/features/html-render/lib/getPageMiddleware.ts]: added

**Review by liximomo — APPROVED** [review]
/lgtm

**Pull request merged**